Addon A11y: Create a11y test provider and revamp a11y addon (#29643)
storybookjs/storybook: code/addons/a11y/README.md

);
```

## Violation impact levels

By default, the addon will consider all violations as errors. However, you can configure the addon to consider some violations as warnings instead of errors. This can be useful when @storybook/addon-a11y is used in combination with `@storybook/experimental-addon-test`. To do this, set the `warnings` parameter in the `a11y` object to an array of impact levels that should be considered as warnings.

```js
export default {
  title: 'button',
  parameters: {
    a11y: {
      /**
       * @default [ ]
       * @type: Array<'minor' | 'moderate' | 'serious' | 'critical'>
       */
      warnings: ['minor', 'moderate'],
    },
  },
};
```

In this example, all violations with an impact level of `minor` or `moderate` will be considered as warnings. All other violations will be considered as errors. When running automated UI tests featured by Vitest, all violations with an impact level of `serious` or `critical` will now fail the test. This failure is reflected as an error in the sidebar or when running Vitest separately. `minor` and `moderate` violations will be reported as warnings but will not fail the test.

> ## Integration with [Component Testing](https://storybook.js.org/docs/writing-tests/component-testing)

Author: ```suggestion
## Integration with [Addon Test](https://storybook.js.org/docs/writing-tests/test-addon)
```

> The a11y addon is compatible with Storybook's newest feature, [component testing](https://storybook.js.org/docs/writing-tests/component-testing). When you run component tests, the a11y addon can automatically check for accessibility issues for all of your stories in the background. If there are any violations, the test will fail, and you will see the results in the sidebar.

Author: ```suggestion
The a11y addon is compatible with Storybook's newest addon, [addon test](https://storybook.js.org/docs/writing-tests/test-addon). When you run component tests, the a11y addon can automatically check for accessibility issues for all of your stories in the background. If there are any violations, the test will fail, and you will see the results in the sidebar.
```

### Automatic setup

> When you add the a11y addon via [add](../api/cli-options.mdx#add) command, it will automatically be integrated with the component testing feature. You don't need to do anything else to enable accessibility checks in your component tests.

Author: ```suggestion
When you add the a11y addon via `npx storybook add` command, it will automatically be integrated with the component testing feature. You don't need to do anything else to enable accessibility checks in your component tests.
```
